src/lambdapi.ml

```diff
@@ -43,15 +48,22 @@ let handle_file : string -> unit = fun fname ->
       | Some(i) -> try with_timeout i compile mp with Timeout ->
                      fatal_no_pos "Compilation timed out for [%s]." fname
     in
-    (* Possibly check the confluence. *)
-    match !confluence_checker with
-    | None      -> ()
-    | Some(cmd) ->
-    let sign = PathMap.find mp Sign.(Timed.(!loaded)) in
-    match Confluence.check cmd sign with
-    | Some(true ) -> ()
-    | Some(false) -> fatal_no_pos "The rewrite system is not confluent."
-    | None        -> fatal_no_pos "The rewrite system may not be confluent."
+    (* [external_checker chk fn kw] verifies if an external checker has been
+       plugged in the variable [chk], if it is the case, it applies the function
+      [fn] to the checker and the signature, and raise an error which contains
+      the keyword [kw]. *)
+    let external_checker chk fn kw =
+      match !chk with
```

Comment:
You can eliminate the first matching using the following scheme.
```ocaml
  let run cmd =
     ... (* body of the Some case *)
  in
  Option.iter run !chk
```
This will limit nesting.